SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -190,6 +191,63 @@ int ASpatialFunctionalTest::GetNumberOfClientWorkers()
 	return Counter;
 }
 
+void ASpatialFunctionalTest::AddActorDelegation_Implementation(AActor* Actor, uint8 ServerWorkerId, bool bPersistOnTestFinished /*= false*/)
+{
+	ISpatialFunctionalTestLBDelegationInterface* DelegationInterface = GetDelegationInterface();
+
+	if (DelegationInterface != nullptr)
+	{
+		bool bAddedDelegation = DelegationInterface->AddActorDelegation(Actor, ServerWorkerId, bPersistOnTestFinished);
+		ensureMsgf(bAddedDelegation, TEXT("Tried to delegate Actor %s to Server Worker %d but couldn't"), *Actor->GetName(), ServerWorkerId);
+	}
+}
+
+void ASpatialFunctionalTest::RemoveActorDelegation_Implementation(AActor* Actor)
+{
+	ISpatialFunctionalTestLBDelegationInterface* DelegationInterface = GetDelegationInterface();
+
+	if (DelegationInterface != nullptr)
+	{
+		bool bRemovedDelegation = DelegationInterface->RemoveActorDelegation(Actor);
+		ensureMsgf(bRemovedDelegation, TEXT("Tried to remove Delegation from Actor %s but couldn't"), *Actor->GetName());
+	}
+}
+
+bool ASpatialFunctionalTest::HasActorDelegation(AActor* Actor, uint8& WorkerId, bool& bIsPersistent)
```

Comment:
By the way, randomly commenting here, but I see WorkerId is uint8, even though SpatialCommonTypes declares it as `using VirtualWorkerId = uint32;`. Maybe we should just switch usages here to VirtualWorkerId? This shouldn't cost a lot of memory, and keeps consistency.

Reply:
The uint8 was there from when I took over the framework. The problem is that uint32 is unsupported as UPROPERTY. How do you feel if I refactor this to int64? Then it's supported by UPROPERTY and supports the whole range of uint32 (not that I think it will be needed).

The other option would be changing VirtualWorkerId to int32. Thoughts?

Reply:
Changed the id to use int32 only in Test related APIs, left VirtualWorkerId as uint32 (although trying to get info from zoning chat if we can change it or not for consistency).

Also tried to use int64 which although it is supported by BP, it's a cumbersome type to use. You can convert int32 to int64 (although it adds an extra node) but you can't do the opposite, so int32 seems like the beast option for now.